### PR TITLE
Add k functionality

### PR DESCRIPTION
**Pull request by fatihertinaz**
Gather functionality from project-k into cd stream

### CODE REVIEW

File: controllers/handler/common.go [hunk 1]
```diff
@@ -20,8 +20,10 @@ import (
 	core "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
 	client "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"time"
```
Comment (horis233) [inline controllers/handler/common.go:27]:
```suggestion
        "time"
      
        operatorv1 "github.com/openshift/api/operator/v1"
	core "k8s.io/api/core/v1"
	"k8s.io/apimachinery/pkg/runtime"
	"k8s.io/apimachinery/pkg/runtime/schema"
	"k8s.io/apimachinery/pkg/util/wait"
	client "sigs.k8s.io/controller-runtime/pkg/client"
	"sigs.k8s.io/controller-runtime/pkg/client/config"
```

File: controllers/handler/deployment.go [hunk 2]
```diff
@@ -307,20 +311,38 @@ func getClusterDomain() (string, error) {
...
-		ServiceName,
-		IAMTokenService,
-		strings.Join([]string{ServiceName, ingressRequest.managementIngress.Namespace}, "."),
-		strings.Join([]string{ServiceName, ingressRequest.managementIngress.Namespace, "svc"}, "."),
-		strings.Join([]string{IAMTokenService, ingressRequest.managementIngress.Namespace}, "."),
-		strings.Join([]string{IAMTokenService, ingressRequest.managementIngress.Namespace, "svc"}, "."),
-	}, " ")
-
-	clusterDomain, err := getClusterDomain()
+
```
Comment (rgtoronto) [inline controllers/handler/deployment.go:316]: remove the empty line on 316

**Review by rgtoronto — COMMENTED** [review]
should we import ``

**Comment by fatihertinaz** [comment]
@horis233 -- Thank you Jiaming!

**Pull request merged**